### PR TITLE
[11/N] Fix extra warnings brought by clang-tidy-17 

### PR DESCRIPTION
**Pull request by cyyever**
Follows #139385

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -35,13 +35,14 @@ cppcoreguidelines-*,
...
 misc-*,
+-misc-confusable-identifiers,
 -misc-const-correctness,
 -misc-include-cleaner,
 -misc-use-anonymous-namespace,
 -misc-unused-parameters,
 -misc-no-recursion,
 -misc-non-private-member-variables-in-classes,
--misc-confusable-identifiers,
+-misc-unused-using-decls,
```
Comment (Skylion007) [inline .clang-tidy:45]: Where are all the problematic unused-using-decls?

File: aten/src/ATen/templates/TensorBody.h [hunk 2]
```diff
@@ -195,7 +195,7 @@ class TORCH_API Tensor: public TensorBase {
   //
   // TODO: temporarily disabled
 
-  Tensor& operator=(const TensorBase& x) & {
+  Tensor& operator=(const TensorBase& x) & noexcept {
```
Comment (Skylion007) [inline aten/src/ATen/templates/TensorBody.h:198]: These are assignment operations, why is it useful they be marked as noexcept?

Comment (Skylion007) [reply aten/src/ATen/templates/TensorBody.h:198]: What about the Scalar assignment ctor down below? Why doesn't that get marked as noexcept too? Bit confused as to why some of these have to be marked as noexcept.

Comment (Skylion007) [reply aten/src/ATen/templates/TensorBody.h:198]: I think this is fine, but I would like to hear the motivation why, I want to make sure it doesn't affect any of the overload for the other assignment operators.

File: .clang-tidy (hunk 1 above)
Comment (cyyever) [reply .clang-tidy:45]: Some headers in ten contain only using statements for used in other cpp files.

File: aten/src/ATen/templates/TensorBody.h (hunk 2 above)
Comment (cyyever) [reply aten/src/ATen/templates/TensorBody.h:198]: Marks them noexcept may help some other classes using them as members to become noexcept in their methods.

**Review by sraikund16 — APPROVED** [review]
profiler fixes look good to me

**Pull request closed without merging**